examples/src/main/java/com/mapbox/navigation/examples/ui/BuildingExtrusionHighlightActivity.kt

```diff
@@ -99,51 +103,99 @@ class BuildingExtrusionActivity : AppCompatActivity(), OnNavigationReadyCallback
+                // Set the Floating Action Buttons to adjust
+                adjust_highlight_color_and_opacity_fab.show()
+                adjust_highlight_color_and_opacity_fab.setOnClickListener {
+                    if (adjustHighlightStyleButtonIndex == opacityList.size) {
+                        adjustHighlightStyleButtonIndex = 0
+                    }
+                    buildingExtrusionHighlightLayer.opacity = opacityList[adjustHighlightStyleButtonIndex]
+                    buildingExtrusionHighlightLayer.color = colorList[adjustHighlightStyleButtonIndex]
+                    adjustHighlightStyleButtonIndex++
                 }
-
-                adjust_building_extrusion_color_and_opacity_fab.setOnClickListener {
-                    if (adjustExtrusionsStyleButtonIndex == opacityList.size) {
-                        adjustExtrusionsStyleButtonIndex = 0
+                adjust_visibility_fab.show()
+                adjust_visibility_fab.setOnClickListener {
+                    mapboxMap.getStyle {
+                        val currentlyVisible = it.getLayer(BuildingExtrusionHighlightLayer.BUILDING_HIGHLIGHTED_EXTRUSION_LAYER_ID)
+                                ?.visibility?.value.equals(Property.VISIBLE)
+                        buildingExtrusionHighlightLayer.updateVisibility(!currentlyVisible)
                     }
-                    buildingExtrusionLayer.opacity = opacityList[adjustExtrusionsStyleButtonIndex]
-                    buildingExtrusionLayer.color = colorList[adjustExtrusionsStyleButtonIndex]
-                    adjustExtrusionsStyleButtonIndex++
                 }
             }
         }
     }
 
-    override fun onRouteProgressChanged(routeProgress: RouteProgress) {
-        // todo
+    override fun onNextRouteLegStart(routeLegProgress: RouteLegProgress) {
+        // Not needed in this example
+    }
+
+    override fun onFinalDestinationArrival(routeProgress: RouteProgress) {
+        mapboxMap.easeCamera(CameraUpdateFactory.zoomTo(18.0), 1800)
+
+        mapboxMap.getStyle {
+
+            /**
+             * Uncomment this commented out code and run the app if you want to add building
+             * extrusions to the map before the device arrives to to the final destination.
+             */
+            /*
+            val buildingFootprintVectorSource = VectorSource(
+                    "building-vector-source", "mapbox://mapbox.mapbox-streets-v8")
+            it.addSource(buildingFootprintVectorSource)
+
+            val fillLayer = FillExtrusionLayer("fill-building-test","building-vector-source")
+            fillLayer.sourceLayer = "building"
+            fillLayer.withProperties(
+                    fillExtrusionColor(Color.GRAY),
+                    fillExtrusionOpacity(.7f),
+                    fillExtrusionHeight(get("height"))
+            )
+            it.addLayer(fillLayer)
+            */
```

Comment:
I am little confused by this comment, because if I want to add building extrusions to the map always just uncommenting this code won't work. I would have to put it somewhere else other than inside `onFinalDestinationArrival`. Also, I feel that rather having this code we should have it covered in our documentation and remove it from above. Wdyt?

Reply:
Sure, I'm good with removing. Had it here to quickly reproduce the effect of extrusions already on the map before `onFinalDestinationArrival()` fires.